### DISCUSSION TITLE
"Breach accounts found" but... nothing on the HIBP website

### DISCUSSION THREAD

**Discussion by jducaud** [dani-garcia/vaultwarden#2376] [Q&A]
### Subject of the issue
My Vaultwarden account email address is said by the Web Vault to be linked to a breach but it does not seem to be.

### Deployment environment
Machine: Synology DS218+ (OS: DSM 6.2.4-25556 Update 5)
Docker image: vaultwarden/server 1.24.0 (latest: 6 weeks ago / 187MB) (Web Vault 2.25.1)
Client: Firefox 98.0.1 (64-bit) (connection to the Web Vault) on Microsoft Windows 10 21H2
Reverse proxy and version (on DSM): nginx 1.16.1

### Steps to reproduce
1 - Log in to my Vaultwarden account on the Web Vault
2 - Go to "Tools > Reports > Data breach report"
3 - Press the "Check breaches" button
4 - The message "BREACHED ACCOUNTS FOUND" (uppercased and red) is displayed, asking for a manual check on HIBP website
5 - Go to HIBP website (just click on the provided hyperlink by the Web Vault)
6 - See that my email address has not been pwned

### Expected behaviour
I have no subscription running at HIBP, so I do not have an API key. I would expect the Web Vault to remind me that I have not an HIBP API key, but without warning me that I have been pwned (this supposed breach has even a date: August 18th 2019).

### Actual behaviour
See above.

### Troubleshooting data
Here are 2 relevant screenshots

Steps 1 to 4
![1 - Vaultwarden](https://user-images.githubusercontent.com/55034588/159129248-a26a2def-4862-49ba-9b78-0f3e0244b3e2.png)

Steps 5 to 6
![2 - HaveIBeenPwned](https://user-images.githubusercontent.com/55034588/159129286-8caafddb-da0d-41f1-8083-de09633e5783.png)

**Comment by BlackDex** [comment]
This is a feature.
Since you do not have a HIBP API-Key you normally would get an error message.
To make it a bit easier for people to check the mail address we added a custom error message noting that the API-Key is not set and we have added a link to HIBP with the mail addresses provided.

Just read the message carefully, and you would see that it states **Manual HIBP Check** and that the Key is not set.

**Comment by jducaud** [comment]
I carefully read the message. Did you carefully read mine?

My point here is to say the Web Vault should not say "there is a breach" in red uppercased letters if the Web Vault is not able to assess the email adress (since no HIBP API key has been provided).

To my mind it is a GUI bug since this message (and I am talking about the "BREACHED ACCOUNTS FOUND" message) is erroneous and scaremongering.

The fact that below this erroneous message there is a (kind) explanation and a hyperlink does not change that. It is actually even worse since a date is provided for the supposed breach (here "18 Aug 2019"), only generating more doubts and confusion.

**Reply by BlackDex** [reply]
That is not something we can't easily fix since this is build-in into the web-vault.
We abuse the response to few a message which makes the web-vault think there is a breach.
There is unfortunately not something we can do about this.

**Comment by jducaud** [comment]
All right, I can hear that. Thank you for your time.

**Comment by DerHary** [comment]
Question would be, if i have a HIBP Key, where to set it??

**Reply by stefan0xC** [reply]
You can [configure](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview) it
https://github.com/dani-garcia/vaultwarden/blob/54bfcb8bc3aa1d15cf20821dc5f3747892011272/.env.template#L183-L185
